framework/encode/vulkan_state_tracker.cpp

```diff
@@ -1352,6 +1350,11 @@ void VulkanStateTracker::TrackUpdateDescriptorSetWithTemplate(VkDescriptorSet
                                              binding.type == VK_DESCRIPTOR_TYPE_SAMPLED_IMAGE ||
                                              binding.type == VK_DESCRIPTOR_TYPE_INPUT_ATTACHMENT;
 
+                if (immutable_image)
```

Comment:
Nit picking: this can be one line `assert((immutable_image && binding.images != nullptr) || (!immutable_image && binding.storage_images != nullptr));` but that's